jlm/hls/backend/rvsdg2rhls/mem-queue.cpp

```diff
@@ -287,10 +284,9 @@ separate_load_edge(
         else
         {
           // end of loop
-          auto load_user_input = dynamic_cast<jlm::rvsdg::simple_input *>(addr_edge_user);
-          JLM_ASSERT(load_user_input);
-          JLM_ASSERT(
-              dynamic_cast<const jlm::hls::branch_op *>(&load_user_input->node()->GetOperation()));
+          auto load_user_input = jlm::util::AssertedCast<jlm::rvsdg::simple_input>(addr_edge_user);
```

Comment:
load_user_input is not used either. They should all be JLM_ASSERTS. This is all just sanity checking.

Reply:
Reverted

Reply:
The dynamic cast is needed for the load_user_input as it is used for the second sanity check. I assume that an AssertedCast becomes the equivalent of a dynamic cast when asserts are disabled.
Ideally, the dynamic cast should also be disabled in a release build.

Reply:
The AssertedCast becomes a static cast when asserts are disabled. Otherwise, it is a dynamic cast.